include/boost/gil/algorithm.hpp

```diff
@@ -440,7 +441,7 @@ void destruct_range_impl(Iterator first, Iterator last,
             std::is_pointer<Iterator>,
             mp11::mp_not
             <
-                std::is_trivially_destructible<typename std::iterator_traits<Iterator>>
+                detail::is_trivially_destructible<typename std::iterator_traits<Iterator>>
```

Comment:
What is the `typename` doing in the above template argument ? It looks wrong...

Reply:
Well spotted! Fixed in separate 5af7a959f8dfa94011037597a9a2a880997dd34b and updated the PR.